### PR TITLE
Enhance/card-section

### PR DESCRIPTION
**Pull request by vitorsilvalima**
Enhanced card section by adding a new strategy that when it receives an icon, it renders the title of the section in a different way

### CODE REVIEW

File: src/Card/CardSection.js [hunk 1]
```diff
@@ -85,6 +129,13 @@ class CardSection extends Component {
         }
       </div>
     )
+
+    return icon ? renderSectionWithIcon() : renderSectionWithoutIcon()
+  }
+
+  renderLineDiviser () {
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Typo: `renderLineDivider`

File: src/Card/CardSection.js [hunk 2]
```diff
@@ -85,6 +129,13 @@ class CardSection extends Component {
...
+    const { theme, icon } = this.props
+    return icon ? (<span className={theme.lineDiviser} />) : null
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Typo: `theme.lineDivider`

Comment (vitorsilvalima) [reply src/Card/CardSection.js]: We are gonna fix all of them even in the css files. Thank you

File: src/Card/CardSection.js [hunk 3]
```diff
@@ -138,6 +191,10 @@ CardSection.propTypes = {
     collapse: PropTypes.element,
     expand: PropTypes.element,
   }),
+  /**
+   * Section Icon that appears in the top left hand corner of the section tittle
+   */
+  icon: PropTypes.element,
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Don't  forget to also add all `theme.<name>` used to `theme` property ☝️

File: src/Card/CardSection.js [hunk 4]
```diff
@@ -67,9 +73,47 @@ class CardSection extends Component {
...
+      }
+    )
+
+    const renderSectionWithIcon = () => (
+      <div className={headerClassNames} {...headerProps}>
+        <div className={theme.sectionInfo}>
+          <span className={theme.sectionIconBox}>
+            {icon}
+          </span>
+          <span className={theme.sectionTitles}>
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Maybe `theme.twoLineTitle` would be a better theme interface?

File: src/Card/CardSection.js [hunk 5]
```diff
@@ -67,9 +73,47 @@ class CardSection extends Component {
...
+      theme.sectionTitle,
+      {
+        [theme.clickableTitle]: onTitleClick,
+        [theme.upperCaseTitles]: !icon,
+      }
+    )
+
+    const renderSectionWithIcon = () => (
+      <div className={headerClassNames} {...headerProps}>
+        <div className={theme.sectionInfo}>
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Maybe `theme.sectionTitleTwoLine` would be a better theme interface?

File: src/Card/CardSection.js [hunk 6]
```diff
@@ -37,7 +49,7 @@ class CardSection extends Component {
 
     return (
       <span className={theme.arrow}>
-        {collapsed ? icons.expand : icons.collapse}
+        {this.getArrowIcon(collapsed)}
```
Comment (derekstavis) [inline src/Card/CardSection.js]: I'd recommend getting `collapsed` from `this.props` inside `getArrowIcon` function, as the icons are being.

File: src/Card/CardSection.js [hunk 7]
```diff
@@ -67,9 +73,47 @@ class CardSection extends Component {
       }
     }
 
-    return (
+    const headerClassNames = classNames(
+      theme.sectionTitle,
+      {
+        [theme.clickableTitle]: onTitleClick,
+        [theme.upperCaseTitles]: !icon,
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Ops! This class doesn't exist anymore!

File: src/Card/CardSection.js [hunk 8]
```diff
@@ -16,18 +17,28 @@ class CardSection extends Component {
...
+  }
+
+  getArrowIcon () {
+    const { collapsed, icons: { expand, collapse } } = this.props
+
+    if (collapsed) {
+      return collapse
+    }
+
+    return expand
```
Comment (derekstavis) [inline src/Card/CardSection.js]: Looks like the logic got inverted here: When collapsed we want to show the expand icon and when expanded, show the collapse one, right?

Comment (vitorsilvalima) [reply src/Card/CardSection.js]: That's right, I removed the flag and than the logic got messed up. I just fixed it by having the icon prop as a flag and returning the right icon according to it

**Pull request merged**